cpp/common/src/codingstandards/cpp/rules/declaredareservedidentifier/DeclaredAReservedIdentifier.qll

```diff
@@ -0,0 +1,37 @@
+/**
+ * Provides a library which includes a `problems` predicate for reporting declarations of reserved identifiers.
+ */
+
+import cpp
+import codingstandards.cpp.Customizations
+import codingstandards.cpp.Exclusions
+import codingstandards.cpp.Naming
+import codingstandards.cpp.CKeywords
+
+abstract class DeclaredAReservedIdentifierSharedQuery extends Query { }
+
+Query getQuery() { result instanceof DeclaredAReservedIdentifierSharedQuery }
+
+query predicate problems(Element m, string message) {
+  not isExcluded(m, getQuery()) and
+  exists(string name |
+    (
+      m.(Macro).hasName(name) or
+      m.(Declaration).hasGlobalName(name)
+    ) and
+    (
+      Naming::Cpp14::hasStandardLibraryMacroName(name)
+      or
+      Naming::Cpp14::hasStandardLibraryObjectName(name)
+      or
+      Naming::Cpp14::hasStandardLibraryFunctionName(name)
```

Comment:
possibly, especially if we are only supporting C99 (or as priority) compared to C11 even,

however there are [some](https://github.com/github/codeql-coding-standards/blob/main/c/misra/src/rules/RULE-21-1/DefineAndUndefUsedOnReservedIdentifierOrMacroName.ql#L31) other [queries](https://github.com/github/codeql-coding-standards/blob/main/c/cert/src/rules/STR32-C/NonNullTerminatedToFunctionThatExpectsAString.ql#L32) where we are also doing this (I know, not the best argument to continue down that road) - I suggest if we want to change this that the effort is orthogonal to this PR, as it will affect many queries, and requires a different dedicated effort to creating a parallel unique lib that contains name sets for alternative standards

Reply:
especially since currently I would not know where to scrape these complete sets from, I know we had suggested [this](https://github.com/p12tic/cppreference-doc/blob/master/index-functions-c.xml) previously, but... what standard version is that?

Reply:
Ok, let's add an issue to this repository to investigate using a different data source for the C standard library names.